src/Libraries/CoreNodes/Data.cs

```diff
@@ -528,6 +529,148 @@ public static bool CanObjectBeCached(object inputObject)
+
+            typeList.Add(polyCurve);
+            typeList.Add(polygon);
+            typeList.Add(rectangle);
+            typeList.Add(new DataNodeDynamoType(typeof(System.DateTime)));
+            typeList.Add(new DataNodeDynamoType(typeof(double), "Number"));
+            typeList.Add(new DataNodeDynamoType(typeof(long), "Integer"));
+            typeList.Add(new DataNodeDynamoType(typeof(Location)));
+            typeList.Add(new DataNodeDynamoType(typeof(Mesh)));
+            typeList.Add(new DataNodeDynamoType(typeof(Plane)));
+            typeList.Add(new DataNodeDynamoType(typeof(Autodesk.DesignScript.Geometry.Point)));
+
+            // Subtypes of Solid
+            var solid = new DataNodeDynamoType(typeof(Solid));
+            var cone = new DataNodeDynamoType(typeof(Cone), solid);    // cone is subtype of solid
+            var cylinder = new DataNodeDynamoType(typeof(Cylinder), cone); // cylinder is subtype of cone 
+            var cuboid = new DataNodeDynamoType(typeof(Cuboid), solid);    // cuboid is subtype of solid
+            var sphere = new DataNodeDynamoType(typeof(Sphere), solid);    // sphere is subtype of solid
+
+            typeList.Add(solid);
+            typeList.Add(cone);
+            typeList.Add(cylinder);
+            typeList.Add(cuboid);
+            typeList.Add(sphere);
+            typeList.Add(new DataNodeDynamoType(typeof(string)));
+
+            // Subtypes of Surface
+            var surface = new DataNodeDynamoType(typeof(Surface));
+            var nurbsSrf = new DataNodeDynamoType(typeof(NurbsSurface), surface);    // nurbsSrf is subtype of surface
+            var polySrf = new DataNodeDynamoType(typeof(PolySurface), surface); // polySrf is subtype of surface
+
+            typeList.Add(surface);
+            typeList.Add(nurbsSrf);
+            typeList.Add(polySrf);
+            typeList.Add(new DataNodeDynamoType(typeof(System.TimeSpan)));
+            typeList.Add(new DataNodeDynamoType(typeof(UV)));
+            typeList.Add(new DataNodeDynamoType(typeof(Vector)));
+
+            return typeList;
+        }
+
+        /// <summary>
+        /// Function to validate input type against supported Dynamo input types
+        /// </summary>
+        /// <param name="inputValue">The incoming data to validate</param>
+        /// <param name="typeString">The input type provided by the user. It has to match the inputValue type</param>
+        /// <param name="isList">The value of this boolean decides if the input is a single object or a list</param>
+        /// <returns></returns>
+        [IsVisibleInDynamoLibrary(false)]
+        public static bool IsSupportedDataNodeDynamoType([ArbitraryDimensionArrayImport] object inputValue, string typeString, bool isList)
+        {
+            var type = GetDataNodeDynamoTypeList().FirstOrDefault(x => x.Type.ToString().Equals(typeString)).Type;
```

Comment:
not great, every time the node' AST is constructed we revaluate and reallocate the list of types, then do string comparisons? I think doing type comparisons will be much faster.